lib/internal/repl.js

```diff
@@ -119,7 +119,7 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
     }
 
     if (data) {
-      repl.history = data.split(/[\n\r]+/, repl.historySize);
+      repl.history = data.split(/[\n\r]+/).slice(repl.historySize * -1);
```

Comment:
I think this code may not be correct though, I think it should probably keep the original logic, and then call `.reverse()`?